eng/scripts/generate_from_source_pom.py

```diff
@@ -220,14 +223,21 @@ def is_track_two_pom(tree_root: ET.Element):
 
 # Creates an artifacts identifier.
 def create_artifact_identifier(element: ET.Element):
+    artifact_id = element_find(element, 'artifactId')
+
+    # not all <dependency> describe maven dependencies
+    # so there might not be an <artifactId>
+    if artifact_id is None:
+        return None
+
```

Comment:
Instead of using this logic, could we make it a bit safer to determine if the xml node is contained within a `dependenciesToScan` xml node, then return None (or another handling)

Reply:
sure, it's not really straightforward though, LMK what you think. What's the concern though? Previously script would explode on dependency without artifactId, now it will ignore it